Programming Course Draft 4.R

```diff
@@ -0,0 +1,433 @@
+
+    min_row <- which(grepl(min, df1_plus_time[ ,2]))
+
+    min_row <- min_row[1]
+
+    # referencing the row number based on the maximum value that the user inputs
+
+    max_row <- which(grepl(max, df1_plus_time[ ,2]))
+
+    max_row <- tail(max_row)
+
+    max_row <- max_row[6]
+
+    # narrowing down the data frame so that it only includes the RR values
+
+    # within the user's specified time frame
+
+    data <- df1_plus_time[min_row:max_row, 1:2]
+
+  }
+
+
+  # warning messages for incorrect data input
+
+  last_minute <- tail(data[,1])
+
+  last_minute <- last_minute[0]
+
+  if (last_minute < .5) {
+
+    warning("This software is not appropriate for heart rate
+            segments smaller than 30 seconds.")
+
+  }
+
+  if (segment = TRUE) {
+
+    if (min > max) {
+
+      warning("Incorrect min/max values. Max value should exceed min value by
+              at least 30 seconds.")
+
+    }
+
+    range <- max - min
+
+    if (max - min < 0.5) {
+
+      warning("This software is not appropriate for heart rate
+            segments smaller than 30 seconds.")
+
+    }
+
+  }
+
+
+  return(data)
+
+}
+
+
+
+#' Part Two: Time-Domain Statistical Methods
+
+#' Root Mean Squared Successive Differences (RMSSD)
+
+#' RMSSD is calculated using the successive differences ellapsed between heart
+#' beats (RR values). This is considered the most precise marker for parasympathetic
+#' activity using heart rate, and it is the most common time-domain HRV statistic reported
+#' in the literature.
+#'
+#' RMSSD values can be subject to individul differences, so it is best to compare RMSSD
+#' values between groups or across time, rather than a single value indicating a validated
+#' level of distress.
+#'
+#' Lower HRV and RMSSD values indicate higher levels of distress.
+#'
+#' @param infile Path to the input file
+#' @return A matrix of the infile
+#' @export
+
+RMSSD_Calculator  <- function(data = data) {
+
+    input_length <- length(data[,2]) - 1
+
+    RR_squared <- data^2
+
+    RR_squared_average <- sum(RR_squared) / input_length
+
+    rmssd1 <- sqrt(RR_squared_average)
+
+    return(rmssd1)
+
+}
+
+
+
+#' SDNN Function
+
+#' SDNN measurement is the standard deviation of all time ellapses heart beats.
+#' This value indicates both parasympathetic and sympathetic nervous system
+#' activity and is imporant measure in cardiology research and practice
+#' It is not often used in psychological research and is most often used in medical research
+#' when heart rate recordings are more than 24 hours. Values below 50ms indicate risk,
+#' while values above 100ms are relativily healthy
+#'
+#' @param infile Path to the input file
+#' @return A matrix of the infile
+#' @export
+
+
+SDNN_Calculation <- function(data) {
+
+  last_minute <- tail(data[,1])
+
+  last_minute <- last_minute[5]
+
+  if (last_minute < 1440) {
+
+    warning("SDNN is most accurate with heart rate segments longer than 24 hours.")
+
+  }
+
+  SDNN <- sd(data[,2])
```

Comment:
i cant tell in github if the data object is named, but if it is using 
data$timeellapses or something like that would make it easier to read for an outsider